port/unix/omrsysinfo.c

```diff
@@ -6638,7 +6769,20 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 		}
 		if (currentElement->isValueToBeChecked) {
 			int64_t result = 0;
-			sscanf(metricElement->value, "%" PRId64, &result);
+			if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE)) {
+				sscanf(metricElement->value, "%" PRId64, &result);
+			} else if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE)) {
+				uint64_t uresult = 0;
+				scanCgroupIntOrMax(portLibrary, metricElement->value, &uresult);
```

Comment:
For cpu.max the problem isn't the multiple items on a line, it is that it is not in key value format. I think we can reuse the key string cpu.max's metricInfoElementList->metricKeyInFile to indicate this format (i.e. NO_KEY) and add logic in iterator_next to read one value at a time (per iterator_next call) from the file string.
I'm reluctant to add new fields to the metric map/metric info element structs since it will be a lot of changes and will only be used by 3 files and only cgroup v2. Would `(OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE) && subsystemMetricMap[state->count]->metricInfoElementList->isValueToBeChecked` work as a condition to find the files with possible max's? isValueToBeChecked seems to indicate that the metric is a limit and for v2 it would only apply to the three .max files.

Reply:
ok, let's take your suggested approach since it requires less changes.